File: AngularApp/angularapp.client/src/app/models/Lote.ts
import { Evento } from "./Evento";

export interface Lote {
  id: number;
  nome: string;
  preco: Float32List;
  dataInicio?: Date;
  dataFim?: Date;
  quantidade: number;
  eventoId: number; 
  evento: Evento; 
}

File: AngularApp/AngularApp.Server/Models/Lote.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace AngularApp.Server.Models
{
    public class Lote
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public decimal Preco { get; set; }
        public DateTime? DataInicio { get; set; }
        public DateTime? DataFim { get; set; }
        public int Quantidade { get; set; }

        [ForeignKey("Evento")]
        public int EventoId { get; set; }
        public Evento Evento { get; set; }
    }
}